### DISCUSSION TITLE
Soft reset feature

### DISCUSSION THREAD

**Discussion by mysuf** [react-hook-form/react-hook-form#6829] [Ideas]
At this moment the only way to reset form is to call `reset(defaultValues)`. It would be nice to have a option to just unlock/clear cache, i.e. forces to re-render and pick defaultValue from each field component again.

Now I have to call `getValues()` to get registered field names and then set new default value to those not dirty, despite the fact that field components have new data available, but it ignores defaultValue attribute.

Worth to say: it shouldn't reset dirty fields. Only those untouched as renderedValue === defaultValue.

**Comment by Evalon** [comment]
Also have such cases, maybe a little bit different. 
For example:
I have a form that consists of dynamic fields loaded from different APIs. 
Every field knows how to load data from api and register the data in form with:
`reset({ ...getValues, [fieldName]: dataFromApi })`
But getValues contains user input and not defaultValues that already has been set in other fields with the same reset call. So, if I reset with getValues I also set user input (if any) as the new default value.
Is it possible to access default values (after reset is called) or apply a soft reset?

**Reply by JamesWong1999** [reply]
does this works for you? `reset(newValue, { keepDefaultValues: true })`

**Reply by Evalon** [reply]
Yeah, It works.
I tried this before. I rejected the use because I found a bug but started to think it was normal behavior.
So the bug is with useWatch that reset to undefined if I use this option.
Thanks for the help! I'm going to create the bug report.

**Reply by Evalon** [reply]
Bug: https://github.com/react-hook-form/react-hook-form/issues/6833

**Reply by Evalon** [reply]
Based on @bluebill1049 response: it's expected behaviour. So soft reset is a feature request.

**Reply by bluebill1049** [reply]
RFC: https://github.com/react-hook-form/react-hook-form/discussions/6884
